moirae/estimators/online/dual.py

```diff
@@ -149,27 +197,51 @@ def initialize_unscented_kalman_filter(
         asoh_wrapper = DegradationModelWrapper(cell_model=cell_model,
                                                asoh=initial_asoh,
                                                transients=initial_transients,
-                                               inputs=initial_inputs)
+                                               inputs=initial_inputs,
+                                               converters={'hidden_conversion_operator': asoh_normalizer})
 
-        # Prepare multivariate Gaussians
+        # Prepare raw multivariate Gaussians
         transients_hidden = convert_vals_model_to_filter(model_quantities=initial_transients,
                                                          uncertainty_matrix=covariance_transient)
         asoh_hidden = MultivariateGaussian(mean=initial_asoh.get_parameters().flatten(),
                                            covariance=covariance_asoh)
         initial_controls = convert_vals_model_to_filter(model_quantities=initial_inputs)
 
+        # Additional setup/conversions
+        trans_proc_noise = transient_covariance_process_noise
+        if trans_proc_noise is not None:
+            trans_proc_noise = cell_wrapper.hidden_conversion.inverse_transform_covariance(
+                transformed_covariance=trans_proc_noise)
+        asoh_proc_noise = asoh_covariance_process_noise
+        if asoh_proc_noise is not None:
+            asoh_proc_noise = asoh_wrapper.hidden_conversion.inverse_transform_covariance(
+                transformed_covariance=asoh_proc_noise)
+        trans_sensor_noise = covariance_sensor_noise
+        asoh_sensor_noise = covariance_sensor_noise
+        if covariance_sensor_noise is not None:
+            trans_sensor_noise = cell_wrapper.output_conversion.inverse_transform_covariance(
+                transformed_covariance=trans_sensor_noise)
+            asoh_sensor_noise = asoh_wrapper.output_conversion.inverse_transform_covariance(
+                transformed_covariance=asoh_sensor_noise)
+
         # Initialize filters
-        trans_filter = UKF(model=cell_wrapper,
-                           initial_hidden=transients_hidden,
-                           initial_controls=initial_controls,
-                           covariance_process_noise=transient_covariance_process_noise,
-                           covariance_sensor_noise=covariance_sensor_noise,
-                           **filter_args.get('transient', UKFTuningParameters.defaults()))
-        asoh_filter = UKF(model=asoh_wrapper,
-                          initial_hidden=asoh_hidden,
-                          initial_controls=initial_controls,
-                          covariance_process_noise=asoh_covariance_process_noise,
-                          covariance_sensor_noise=covariance_sensor_noise,
-                          **filter_args.get('asoh', UKFTuningParameters.defaults()))
+        trans_filter = UKF(
```

Comment:
Tests which ensure we didn't have these numerous conversions backwards would be awesome

Reply:
Good idea! Will write one as well!